### PR TITLE
Bootstrap Toolbar and Simulation Logic

### PR DESCRIPTION
**Pull request by dpasillas**
# What does this code do?
Bootstrapped Toolbar with simulation controls
Added simulation controls to the board
Made some changes to remove performance bottlenecks.

# What testing did you do?
- Launched locally to verify toolbar was present. 
- Hovered and clicked toolbar buttons to verify styles were set correctly.
- Used simulation controls to verify that the simulation time was running properly
- Ran simulation after changes to verify performance bottlenecks were removed.

# Meta
#13

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -15,25 +16,37 @@ interface IState {}
...
+  private parts: Part[] = [
+    new Part({type: PartType.GATE, subtype: GateType.AND, label:"AND"}),
+    new Part({type: PartType.GATE, subtype: GateType.NAND, label:"NAND"}),
+    new Part({type: PartType.GATE, subtype: GateType.OR, label:"OR"}),
+    new Part({type: PartType.GATE, subtype: GateType.NOR, label:"NOR"}),
+    new Part({type: PartType.GATE, subtype: GateType.XOR, label:"XOR"}),
+    new Part({type: PartType.GATE, subtype: GateType.XNOR, label:"XNOR"}),
+    new Part({type: PartType.GATE, subtype: GateType.BUF, label:"BUF"}),
+    new Part({type: PartType.GATE, subtype: GateType.NOT, label:"NOT"}),
+    new Part({type: PartType.OUTPUT, subtype: 0, label:"Bulb"}),
```
Comment (svtanthony) [inline src/components/App.tsx:28]: add new switch

File: src/logic/LogicGate.tsx [hunk 2]
```diff
@@ -41,7 +41,24 @@ function isNot(type: GateType) {
     case GateType.BUF:
       return false;
     default:
-      throw Error("Unsupported Gate Type")
+      throw new Error(`Unsupported Gate Type(${type})`)
+  }
+}
+
+function maxPins(type: GateType): number {
```
Comment (svtanthony) [inline src/logic/LogicGate.tsx:48]: how will this work if we have dynamic resizing (width, etc)?

**Review by svtanthony — APPROVED** [review]

File: src/logic/LogicGate.tsx (hunk 2 above)
Comment (dpasillas) [reply src/logic/LogicGate.tsx:48]: We'll filter in the UI when these properties are changed, and we may also filter when these values are changed.  For now, width and fieldWidth are accessible and modifiable as properties.

File: src/components/App.tsx (hunk 1 above)
Comment (dpasillas) [reply src/components/App.tsx:28]: I'll add when resolving merge conflicts

**Pull request merged**